src/gcp_scanner/scanner.py

```diff
@@ -269,12 +271,20 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       sa_results_data = json.dumps(sa_results, indent=2, sort_keys=False)
 
-      with open(out_dir + '/%s.json' % project_id, 'a',
-                encoding='utf-8') as outfile:
-        outfile.write(sa_results_data)
-
-      # Clean memory to avoid leak for large amount projects.
-      sa_results.clear()
+      # Generate current timestamp to append to output filename
+      scan_time_suffix = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
```

Comment:
Hmm, that could be confusing for a user since this code block is executed for each project in the scan pipeline. So, timestamp will be different for all of them. I'd suggest to have `scan_time_suffix` to be the same for all projects in the scan batch.

Reply:
Good point. In fact, it's probably better if the output file is opened at the very start since this also makes sure the output path is valid.

Reply:
Implemented the suggestions :)